tests/unit/openstack/test_environment.py

```diff
@@ -178,3 +226,49 @@ def test_from_deployment_location_auth_collects_properties_with_prefix(self):
         self.assertIn('domain_id', openstack_auth.auth_properties)
         self.assertEqual(openstack_auth.auth_properties['domain_id'], 'testdomain')
         self.assertEqual(len(openstack_auth.auth_properties), 3)
+
+    def test_from_deployment_location_with_certs(self):
+        translator = OpenstackDeploymentLocationTranslator()
+        certs_dir = os.path.dirname(os.path.abspath(certs.__file__))
+        ca_cert_path = os.path.join(certs_dir, 'ca.cert')
+        with open(ca_cert_path, 'r') as f:
+            ca_cert = f.read()
+        client_cert_path = os.path.join(certs_dir, 'client.cert')
+        with open(client_cert_path, 'r') as f:
+            client_cert = f.read()
+        client_key_path = os.path.join(certs_dir, 'client.key')
+        with open(client_key_path, 'r') as f:
+            client_key = f.read()
+        openstack_location = translator.from_deployment_location({'name': 'testdl', 'properties': {
+            OS_URL_PROP: 'testip',
+            AUTH_API_PROP: 'identity/v3',
+            'os_auth_username': 'test',
+            'os_auth_password': 'secret',
+            'os_auth_domain_id': 'testdomain',
+            'os_cacert': ca_cert,
+            'os_cert': client_cert, 
+            'os_key': client_key
+        }})
+        self.assertEqual(openstack_location._OpenstackDeploymentLocation__ca_cert, ca_cert)
+        self.assertEqual(openstack_location._OpenstackDeploymentLocation__client_cert, client_cert)
+        self.assertEqual(openstack_location._OpenstackDeploymentLocation__client_key, client_key)
```

Comment:
Does this need an openstack_location.close(), to clean up?

Reply:
Because we don't create the session the certs are not written, so close is unnecessary